docs/operator-manual/upgrading/2.14-3.0.md

```diff
@@ -209,6 +209,50 @@ Explanation:
 - In Helm 3.17.1, this behavior changed and `null` object in `values.yaml` still results in this warning upon performing `helm template`, but the resulting K8s object will be overridden with the invalid `null` value.
 - To resolve the issue, identify `values.yaml` with `null` object values, and remove those `null` values.   
 
+### Use Annotation-Based Tracking by Default
+
+The default behavior for [tracking resources](../../user-guide/resource_tracking.md) has changed to use annotation-based 
+tracking instead of label-based tracking. Annotation-based tracking is more reliable and less prone to errors caused by 
+external code copying tracking labels from one resource to another.
+
+#### Detection
+
+To detect if you are impacted, check the `argocd-cm` ConfigMap for the `application.resourceTrackingMethod` field. If it 
+unset or is set to `label`, you are using label-based tracking. If it is set to `annotation`, you are already using
+annotation-based tracking and are not impacted by this change.
+
+```sh
+kubectl get cm argocd-cm -n argocd -o jsonpath='{.data.application\.resourceTrackingMethod}'
+```
+
+#### Remediation
+
+For most users, it is safe to upgrade to Argo CD 3.0 and use annotation-based tracking. Labels will be replaced with
+annotations on the next sync. Applications will not be marked as out-of-sync if the labels are not present on the
+resources.
```

Comment:
In the past, a problem with changing the tracking annotation (from label to label+annotations) was that, even though apps were still "synced" (no drift), if the Application or any of the resources is deleted by the next sync, Argo would actually orphan these resources instead of deleting them. 

I think we should add a `!!! warning` for that scenario. In the past, triggering a sync on all Application fixed the issues, but that might 1) Not always be possible due to sync windows or other user processes and 2) If the sync failed, the failed resources might still be missing the annotation.

Reply:
Added a warning, PTAL